src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -13484,7 +13484,8 @@ bool Unit::UpdatePosition(float x, float y, float z, float orientation, bool tel
         return false;
     }
 
-    bool const turn = (GetOrientation() != orientation);
+    // Check if angular distance changed
+    bool const turn = M_PI - fabs(fabs(GetOrientation() - orientation) - M_PI) > 0.001f;
```

Comment:
Checking orientation cannot be made with simple subtraction, as we have to compare angular distance. This formula will leave result in range of 0..PI (no two points can have greater distance than PI). I tried to search for similar formula in G3D but failed to do so. Advise required.